reactor-core/src/main/java/reactor/core/publisher/FluxPublish.java

```diff
@@ -483,8 +484,8 @@ final void drain() {
 				}
 				else if (sourceMode == Fuseable.SYNC) {
 					done = true;
-					if (checkTerminated(true, true)) {
-						return;
+					if (checkTerminated(true, empty)) {
+						break;
```

Comment:
this can still be return ?

Reply:
isn't it more correct to use break, allowing a racing WIP to be picked up and making the code more future-proof? (eg. if at some point we implement doOnDiscard and the queue is fed one element in race with a cancel)

Reply:
you're right it doesn't change anything the `WIP.addAndGet` is inside the loop